src/pages/api/sfcc/warranty.ts

```diff
@@ -0,0 +1,91 @@
+        }
+    ]
+  };
+
+  const prepareImageData = (imageData : AttachmentData, parentId : string): AttachmentData => {
+    const records = imageData.records.map(record => ({
+      ...record,
+      parentId: parentId
+    }));
+
+    return {
+      ...imageData,
+      records,
+    };
+  }
```

Comment:
why not just do this on the fly in the final payload, there's no crazy logic here that needs to be abstracted into a function

Reply:
This is a common code block of preparing image upload btw contact us and warranty page.
Ultimately both the pages and all routes will leverage this image data prepration abstraction leaving the main page(warranty.ts  and contactus.ts) lean and clean.